cylc/flow/tests/test_platform_lookup.py

```diff
@@ -0,0 +1,109 @@
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) 2008-2019 NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Tests for the platform lookup.
+
+import pytest
+from cylc.flow.platform_lookup import forward_lookup, reverse_lookup
+from cylc.flow.exceptions import PlatformLookupError
+
+# The platforms list for testing is set as a constant
+# [platforms]
+#     [[desktop\d\d|laptop\d\d]]
+#         # hosts = platform name (default)
+#         # Note: "desktop01" and "desktop02" are both valid and distinct
+#                  platforms
+#     [[sugar]]
+#         hosts = localhost
+#         batch system = slurm
+#     [[hpc]]
+#         hosts = hpcl1, hpcl2
+#         retrieve job logs = True
+#         batch system = pbs
+#     [[hpcl1-bg]]
+#         hosts = hpcl1
+#         retrieve job logs = True
+#         batch system = background
+#     [[hpcl2-bg]]
+#         hosts = hpcl2
+#         retrieve job logs = True
+#         batch system = background
```

Comment:
Can remove this development comment?

Reply:
We can if you want
But
This is meant to show the sort of file parsed to create the dict below.